### PR TITLE
Refactor MessageComposerButtons

### PR DESCRIPTION
**Pull request by andybalaam**
Should not affect behaviour

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

Preview: https://61f7a7eb9823c9751356b2a6--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/rooms/MessageComposerButtons.tsx [hunk 1]
```diff
@@ -270,6 +270,51 @@ class UploadButton extends React.Component<IUploadButtonProps> {
...
+                className="mx_MessageComposer_button mx_MessageComposer_stickers"
+                onClick={() => props.setStickerPickerOpen(!props.isStickerPickerOpen)}
+                title={
+                    props.narrowMode
+                        ? null
+                        : props.isStickerPickerOpen
+                            ? _t("Hide Stickers")
+                            : _t("Show Stickers")
+                }
+                label={props.narrowMode ? _t("Send a sticker") : null}
```
Comment (turt2live) [inline src/components/views/rooms/MessageComposerButtons.tsx:289]: it's a little surprising that we disable the labels and such on narrow mode, but don't do the same to other buttons.

**Review by turt2live — APPROVED** [review]
overall seems fine. The diff is a bit difficult to read and the purpose isn't entire clear, but the code does appear to be equivalent. 

The nested ternaries make me extremely nervous, though I don't think they're banned by code style. Wrapping the nested ones in brackets would add a little bit of comfort.

File: src/components/views/rooms/MessageComposerButtons.tsx (hunk 1 above)
Comment (andybalaam) [reply src/components/views/rooms/MessageComposerButtons.tsx:289]: Yes, but this change doesn't change anything, so I won't modify that here.

**Comment by andybalaam** [comment]
> overall seems fine. The diff is a bit difficult to read and the purpose isn't entire clear, but the code does appear to be equivalent.

Sorry, yeah I should have explained the purpose more.  I am working on a task to change the layout of the buttons in the composer, and the purpose of this change is to improve the code so actually making the functional change is straightforward and easy to understand.

The fact that the diff is difficult to read is the reason why I wanted to separate the refactoring from the functional change.

> The nested ternaries make me extremely nervous, though I don't think they're banned by code style. Wrapping the nested ones in brackets would add a little bit of comfort.

Hmm... they are supposed to make it clearer what the code is doing, but maybe good old-fashioned `if` would be better here.

**Pull request merged**